decoder.go

```diff
@@ -259,7 +291,7 @@ func (d *decoder) decode(nalu []byte) (*avFrameWrapper, error) {
 		// This allocates the underlying byte array to contain the image data.
 		res = C.av_frame_get_buffer(dst.frame, 1)
 		if res < 0 {
-			return nil, errors.New("av_frame_get_buffer() err")
+			return nil, newAvError(res, "av_frame_get_buffer() err")
```

Comment:
[q] Should we make this recoverable? Could there be some scenario where unable to allocate memory temporarily?

Reply:
That makes sense to me. Could happen in memory pressure scenarios where we'd want to recover

Reply:
Done